### DISCUSSION TITLE
<nuxt-link> triggers useAsyncData requests on static site

### DISCUSSION THREAD

**Discussion by marcodeluca** [nuxt/nuxt#28020] [Questions]
Hello - I'll keep this brief unless someone who's available to help wants more info. I have a nuxt 3 site and experiencing some strange behaviour:

`<nuxt-link> ` is triggering useAsyncData requests on the static generated site, while `<a href>` works as expected. My dynamic page loads data like this:

```
const { data, pending, error } = await useAsyncData('silos', async () => {
  return $fetch(`https://url.com/api/silos?filters[slug]=${slug}&populate=deep`);
});
```

Data is coming from Strapi.

using npx nuxi generate, i get my static site with all my pages and data working fine. If I put a url into the browser, page loads fine. If I have the following in my menu and click on the link, it works fine:

`<a href="/page-slug">Page Title</a>`

However, if I use this in my menu:

`<nuxt-link to="/page-slug">Page Title</nuxt-link>`

The async request is run, and if I have strapi turned off, i get a 502 error because strapi is off.

Any reason why nuxt-link does this?

**Comment by pixelpaulaus** [comment]
did you ever resolve this?

**Comment by AlexMili** [comment]
I have exactly the same issue ! More specifically, when my API is down, its details are leaked in the console.

@marcodeluca were you able to fix it?

**Comment by marcodeluca** [comment]
@AlexMili - nope but I didnt really spend time trying. We just changed our project files to use regular links. 

We've been using strapi + nuxt for a year now and in general we've found that as long as we stay away from most plugins things have been good and fairly stable. I find nuxt a bit... overly complex I think in some instances when the goal is to end up with a static site, especially using a data source like Strapi vs having all of your content in nuxt.

Dealing with plugins like nuxt-link and nuxt-img had too many issues or specific requirements that we just stay away from them.

Sorry to be not helpful:)

**Reply by AlexMili** [reply]
Thank you for your answer ! I might do something similar even if there is the possibility to use Nitro server as gateway when not entirely in static (which is my case) https://github.com/nuxt/nuxt/issues/27840.
